### PR TITLE
Fix appveyor ci for round3

### PR DESCRIPTION
**Pull request by cosmo0920**
**Which issue(s) this PR fixes**: 

Related to #2854.

* Omit deleting ephemeral buffer file due to permission error.
  * Windows does not permit removing files which are used in another process even if administrator.
* Fix `linger_timeout` handling on Windows

**What this PR does / why we need it**: 
We should take care of AppVeyor CI result.

**Docs Changes**:
No needed.

**Release Note**: 

None.

### CODE REVIEW

File: lib/fluent/plugin_helper/socket_option.rb [hunk 1]
```diff
@@ -49,9 +51,21 @@ def socket_option_set(sock, resolve_name: nil, nonblock: false, linger_timeout:
...
         end
-        if linger_timeout
+        if Fluent.windows?
+          # To prevent closing socket forcibly on Windows,
+          # this options shouldn't be set up when linger_timeout equals to 0 (including nil).
+          # This unintended behavior always ocurrs on Windows when linger_timeout.to_i == 0.
+          # This unintented behavior causes "Errno::ECONNRESET: An existing connection was forcibly
+          # closed by the remote host." on Windows.
+          if linger_timeout.to_i > 0
+            optval = [1, linger_timeout.to_i].pack(FORMAT_STRUCT_LINGER_WINDOWS)
```
Comment (ganmacs) [inline lib/fluent/plugin_helper/socket_option.rb]: It may be better to check `linger_timeout` should be less than 2^16? or this case won't happen?

Comment (cosmo0920) [reply lib/fluent/plugin_helper/socket_option.rb]: Hmm..., specifying `2**17` here does not cause `Errno::ECONNRESET`.
But it maybe reasonable to check greater than 65535(2**16).

Comment (cosmo0920) [reply lib/fluent/plugin_helper/socket_option.rb]: Fixed in https://github.com/fluent/fluentd/pull/2862/commits/b55e610909943cce71dda04c242a87f573fc1be9

**Comment by cosmo0920** [comment]
I'll merge this PR after finishing CIs.

**Pull request merged**